okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -127,64 +127,55 @@ public static void getLatestProducts(int limit, List<ModuleDescriptor> mdl) {
   }
 
   /**
-   * Check if md's required and optional interfaces are provided by set of modules.
-   * The interface version is not checked, only the interface name.
-   * Only interfaces listed in allProvided are checked, this allows to exclude some optional
-   * interfaces from the check.
+   * Check if md's required interfaces are provided by set of modules.
    *
-   * @param modules     the modules that check against
-   * @param allProvided interfaces for all modules
+   * @param modules     the modules to check against
    * @param md          module to check.
-   * @return            true if interface requirements are met
+   * @return            true if interface requirements are met; false otherwise
    */
-  static boolean moduleDepProvided(
-      List<ModuleDescriptor> modules, Set<String> allProvided,
-      ModuleDescriptor md) {
+  public static boolean moduleDepProvided(
+      Collection<ModuleDescriptor> modules, ModuleDescriptor md) {
 
-    for (InterfaceDescriptor req : md.getRequiresOptionalList()) {
-      if (allProvided.contains(req.getId())) {
-        boolean found = false;
-        for (ModuleDescriptor md1 : modules) {
-          InterfaceDescriptor[] providesList = md1.getProvidesList();
-          for (InterfaceDescriptor prov : providesList) {
-            if (prov.isRegularHandler() && prov.getId().equals(req.getId())) {
-              found = true;
-            }
+    interfaceDescriptor:
+    for (InterfaceDescriptor req : md.getRequiresList()) {
+      for (ModuleDescriptor md1 : modules) {
+        InterfaceDescriptor[] providesList = md1.getProvidesList();
+        for (InterfaceDescriptor prov : providesList) {
+          if (prov.isRegularHandler() && prov.isCompatible(req)) {
+            continue interfaceDescriptor;
           }
         }
-        if (!found) {
-          return false;
-        }
       }
+      return false;
     }
     return true;
   }
 
-  static void topoSort(List<ModuleDescriptor> modules) {
-    List<ModuleDescriptor> result = new LinkedList<>();
+  /**
+   * Check if any of md's provided interfaces are used by any of the modules.
+   * <p>A interface is used if it's a required or optional interface dependency.</p>
```

Comment:
The closing p HTML tag is not required because javadoc uses HTML, not XHTML.
It is very uncommon to use the closing p HTML tag. Therefore some style guides explicitly disallow `</p>`:
https://blog.joda.org/2012/11/javadoc-coding-standards.html
https://metaterasology.github.io/docs/workflow/documentationJavaDoc.html#html-tags
Using `</p>` is uncommon in FOLIO javadoc and Okapi javadoc.

Reply:
ok.